behaviour/graql/language/delete.feature

```diff
@@ -649,7 +977,284 @@ Feature: Graql Delete Query
+      | pcW1 | value | postcode:W1U8ED |
+    Then get answers of graql query
+      """
+      match $x has attribute $a; get;
+      """
+    Then uniquely identify answer concepts
+      | x    | a    |
+      | SHER | nSLK |
+      | SHER | pcW1 |
+    Then graql delete
+      """
+      match
+        $x isa person, has attribute $a;
+        $a isa postcode;
+      delete
+        $x has attribute $a;
+      """
+    Then the integrity is validated
+    Then get answers of graql query
+      """
+      match $x has attribute $a; get;
+      """
+    Then uniquely identify answer concepts
+      | x    | a    |
+      | SHER | nSLK |
+
+
+  Scenario: an attribute ownership can be deleted using its supertype as a label
+    Given graql define
+      """
+      define
+      address sub attribute, value string;
+      postcode sub address;
+      person has postcode;
+      """
+    Given the integrity is validated
+    When graql insert
+      """
+      insert
+      $x isa person, has name "Sherlock", has postcode "W1U8ED";
+      """
+    When the integrity is validated
+    Then concept identifiers are
+      |      | check | value           |
+      | SHER | key   | name:Sherlock   |
+      | nSLK | value | name:Sherlock   |
+      | pcW1 | value | postcode:W1U8ED |
+    Then get answers of graql query
+      """
+      match $x has address $a; get;
+      """
+    Then uniquely identify answer concepts
+      | x    | a    |
+      | SHER | pcW1 |
+    Then graql delete
+      """
+      match
+        $x isa person, has address $a;
+      delete
+        $x has address $a;
+      """
+    Then the integrity is validated
+    Then get answers of graql query
+      """
+      match $x has address $a; get;
+      """
+    Then answer size is: 0
+
+
+  Scenario: delete an attribute ownership using too-specific (downcasting) type throws
+    Given graql define
+      """
+      define
+      address sub attribute, value string;
+      postcode sub address;
+      person has address;
+      """
+    Given the integrity is validated
+    When graql insert
+      """
+      insert
+      $x isa person, has name "Sherlock", has address "221B Baker Street";
+      """
+    When the integrity is validated
+    Then graql delete
+      """
+      match
+        $x isa person, has address $a;
+      delete
+        $x has postcode $a;
+      """
+
+
+  Scenario: deleting an attribute ownership using 'thing' as a label throws an error
+    Given graql define
+      """
+      define
+      address sub attribute, value string;
+      postcode sub address;
+      person has postcode;
+      """
+    Given the integrity is validated
+    When graql insert
+      """
+      insert
+      $x isa person, has name "Sherlock", has postcode "W1U8ED";
+      """
+    When the integrity is validated
+    Then graql delete throws
+      """
+      match
+        $x isa person, has address $a;
+      delete
+        $x has thing $a;
+      """
```

Comment:
For the delete throws case we definitely want to check integrity afterwards IMO

Reply:
Done